src/database/schema/ChannelListSchema.ts

```diff
@@ -515,14 +516,17 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data?.targetImage || DEFAULT_PROFILE_PIC_PATH,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for handling cases where `targetImage` is not available. However, it's important to note that the `channelType` is now dynamic and can be either 'ANON_PM' or 'PM'. Make sure this doesn't introduce any unexpected behavior in other parts of your codebase that rely on the `channelType` being strictly 'ANON_PM'.
```diff
- channelType: 'ANON_PM',
+ channelType: type,
```
Also, you have removed the `lastUpdatedAt`, `lastUpdatedBy`, and `createdAt` fields from the new object creation. If these fields are necessary for your application, please add them back.
```diff
  channelType: type,
+ lastUpdatedAt: data?.message?.created_at,
+ lastUpdatedBy: data?.message?.user?.id,
+ createdAt: data?.message?.created_at,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->